packages/ledgerstate/objectstorage.go

```diff
@@ -38,66 +40,100 @@ const (
+	outputStorageOptions []objectstorage.Option
 
-// consumerStorageOptions contains a list of default settings for the OutputMetadata object storage.
-var consumerStorageOptions = []objectstorage.Option{
-	ConsumerPartitionKeys,
-	objectstorage.CacheTime(10 * time.Second),
-	objectstorage.LeakDetectionEnabled(false),
-}
+	// outputMetadataStorageOptions contains a list of default settings for the OutputMetadata object storage.
+	outputMetadataStorageOptions []objectstorage.Option
+
+	// consumerStorageOptions contains a list of default settings for the OutputMetadata object storage.
+	consumerStorageOptions []objectstorage.Option
+
+	// addressOutputMappingStorageOptions contains a list of default settings for the AddressOutputMapping object storage.
+	addressOutputMappingStorageOptions []objectstorage.Option
+)
 
-// addressOutputMappingStorageOptions contains a list of default settings for the AddressOutputMapping object storage.
-var addressOutputMappingStorageOptions = []objectstorage.Option{
-	objectstorage.CacheTime(10 * time.Second),
-	objectstorage.PartitionKey(AddressLength, OutputIDLength),
-	objectstorage.LeakDetectionEnabled(false),
+func initObjectStorageOptions(cacheProvider *database.CacheTimeProvider) {
+	branchStorageOptions = []objectstorage.Option{
+		cacheProvider.CacheTime(branchCacheTime),
+		objectstorage.LeakDetectionEnabled(false),
+	}
+
+	childBranchStorageOptions = []objectstorage.Option{
+		ChildBranchKeyPartition,
+		cacheProvider.CacheTime(branchCacheTime),
+		objectstorage.LeakDetectionEnabled(false),
+	}
+
+	conflictStorageOptions = []objectstorage.Option{
+		cacheProvider.CacheTime(consumerCacheTime),
+		objectstorage.LeakDetectionEnabled(false),
+	}
+
+	conflictMemberStorageOptions = []objectstorage.Option{
+		ConflictMemberKeyPartition,
+		cacheProvider.CacheTime(conflictCacheTime),
+		objectstorage.LeakDetectionEnabled(false),
+	}
+
+	transactionStorageOptions = []objectstorage.Option{
+		cacheProvider.CacheTime(transactionCacheTime),
+		objectstorage.LeakDetectionEnabled(false),
+	}
+
+	transactionMetadataStorageOptions = []objectstorage.Option{
+		cacheProvider.CacheTime(transactionCacheTime),
+		objectstorage.LeakDetectionEnabled(false),
+	}
+
+	outputStorageOptions = []objectstorage.Option{
+		cacheProvider.CacheTime(outputCacheTime),
+		objectstorage.LeakDetectionEnabled(false),
+	}
+
+	outputMetadataStorageOptions = []objectstorage.Option{
+		cacheProvider.CacheTime(outputCacheTime),
+		objectstorage.LeakDetectionEnabled(false),
+	}
+
+	consumerStorageOptions = []objectstorage.Option{
+		ConsumerPartitionKeys,
+		cacheProvider.CacheTime(consumerCacheTime),
+		objectstorage.LeakDetectionEnabled(false),
+	}
+
+	addressOutputMappingStorageOptions = []objectstorage.Option{
+		cacheProvider.CacheTime(addressCacheTime),
+		objectstorage.PartitionKey(AddressLength, OutputIDLength),
+		objectstorage.LeakDetectionEnabled(false),
+	}
```

Comment:
question:
In go we call non-exported variable "global" still?  How would we call exported ones?

The main motivation for the change, as I understand, is for tests correct (even though you can argue it adds flexibility to the prod code)?

Reply:
Sorry, I might misuse the term global variable, let's just call it a package scoped variable. Of course, when it's unexported the situation isn't that bad, but it doesn't solve the original problem: that you have a shared state and it's get mutated for no reason. It's not only related to tests, in production code multiple callers might call our public API concurrently and you will have a data race condition. So again, we should avoid such singleton-ish instances in packages.

Reply:
yeah I agree
I initally mapped the concept of go package in my head to the concept of class from java (because this is what it looked like to me from the code)

but the way you think about it sounds better and more proper